src/main/java/net/openhft/chronicle/network/connection/AbstractConnectionStrategy.java

```diff
@@ -0,0 +1,122 @@
+package net.openhft.chronicle.network.connection;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.network.AlwaysStartOnPrimaryConnectionStrategy;
+import net.openhft.chronicle.network.ConnectionStrategy;
+import net.openhft.chronicle.network.VanillaClientConnectionMonitor;
+import net.openhft.chronicle.wire.AbstractMarshallableCfg;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.*;
+import java.util.Enumeration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static net.openhft.chronicle.network.connection.TcpChannelHub.TCP_BUFFER;
+
+public abstract class AbstractConnectionStrategy extends AbstractMarshallableCfg implements ConnectionStrategy {
+    protected int tcpBufferSize = Jvm.getInteger("tcp.client.buffer.size", TCP_BUFFER);
+    private final transient AtomicBoolean isClosed = new AtomicBoolean(false);
+    protected ClientConnectionMonitor clientConnectionMonitor = new VanillaClientConnectionMonitor();
+    protected String localSocketBindingHost;
+    protected int localSocketBindingPort = 0;
```

Comment:
Perhaps having binding port is an overkill, I've never seen any software specifying binding port for outgoing connections and not confident it will actually work in practice, though you never know...